docs/docs/reference/release-notes/migrating-from-v2-to-v3.md

```diff
@@ -0,0 +1,588 @@
+
+The APIs `push`, `replace` & `navigateTo` in `gatsby-link` (an internal package) were deprecated in v2 and now with v3 completely removed. Please use `navigate` instead.
+
+```diff
+import React from "react"
+- import { navigateTo, push, replace } from "gatsby"
++ import { navigate } from "gatsby"
+
+const Form = () => (
+  <form
+    onSubmit={event => {
+      event.preventDefault()
+
+-     navigateTo("/form-submitted/") // or push() or replace()
++     navigate("/form-submitted/")
+    }}
+  >
+)
+```
+
+### Removal of `__experimentalThemes`
+
+The deprecated `__experimentalThemes` key inside `gatsby-config.js` was removed. You'll need to define your Gatsby themes inside the `plugins` array like any other plugin.
+
+```diff:title=gatsby-config.js
+module.exports = {
+- __experimentalThemes: ["gatsby-theme-blog"]
++ plugins: ["gatsby-theme-blog"]
+}
+```
+
+### Removal of `pathContext`
+
+The deprecated API `pathContext` was removed. You need to rename instances of it to `pageContext`. For example, if you passed information inside your `gatsby-node.js` and accessed it in your page:
+
+```diff:title=src/templates/context.jsx
+import React from "react"
+
+- const ContextPage = ({ pathContext }) => (
++ const ContextPage = ({ pageContext }) => (
+  <main>
+    <h1>Hello from a page that uses the old pathContext</h1>
+    <p>It was deprecated in favor of pageContext</p>
+-   <p>{pathContext.foo}</p>
++   <p>{pageContext.foo}</p>
+  </main>
+)
+
+export default ContextPage
+```
+
+### Removal of `boundActionCreators`
+
+The deprecated API `boundActionCreators` was removed. Please rename its instances to `actions` to keep the same behavior. For example, in your `gatsby-node.js` file:
+
+```diff:title=gatsby-node.js
+exports.createPages = (gatsbyArgs, pluginArgs) => {
+- const { boundActionCreators } = gatsbyArgs
++ const { actions } = gatsbyArgs
+}
+```
+
+### Removal of `deleteNodes`
+
+The deprecated API `deleteNodes` was removed. Please iterate over the `nodes` yourself and call `deleteNode`:
+
+```diff
+const nodes = ["an-array-of-nodes"]
+- deleteNodes(nodes)
++ nodes.forEach(node => deleteNode(node))
+```
+
+### Removal of `fieldName` & `fieldValue` from `createNodeField`
+
+The arguments `fieldName` and `fieldValue` were removed from the `createNodeField` API. Please use `name` and `value` instead.
+
+```diff:title=gatsby-node.js
+exports.onCreateNode = ({ node, actions }) => {
+  const { createNodeField } = actions
+
+  createNodeField({
+    node,
+-   fieldName: "slug",
+-   fieldValue: "my-custom-slug",
++   name: "slug",
++   value: "my-custom-slug",
+  })
+}
+```
+
+### Removal of `hasNodeChanged` from public API surface
+
+The check whether a node has changed or not is already happening internally so it's not necessary to use this API in your plugins anymore.
+
+### Removal of `sizes` & `resolutions` for image queries
+
+The `sizes` and `resolutions` queries were deprecated in v2 in favor of `fluid` and `fixed`.
```

Comment:
Should we add a note here about the whole plugin being deprecated and we recommend migration?